drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py

```diff
@@ -0,0 +1,758 @@
+    new_ngs_sample.setParentSampleIdentifiers([tumor_dna_sample_id])
+    new_ngs_sample.setExperiment(new_ngs_experiment)
+
+    if not barcode[0] in basename:
+        parent_dir_path = os.path.dirname(in_file)
+        print(parent_dir_path)
+        new_path = os.path.join(parent_dir_path, '{}_{}'.format(tumor_dna_sample_id, basename))
+        print(new_path)
+        os.rename(in_file, new_path)
+        in_file = new_path
+
+    # Create a data-set attached to the VARIANT CALL sample
+    data_set = transaction.createNewDataSet("Q_NGS_VARIANT_CALLING_DATA")
+    data_set.setMeasuredData(False)
+    data_set.setSample(new_ngs_sample)
+
+    # Attach the directory to the dataset
+    transaction.moveFile(in_file, data_set)
+
+    # Update sample location
+    update_sample_location_to_qbic(tumor_dna_sample_id)
+
+
+def find_pbmc(in_file, transaction):
+    basename = os.path.basename(in_file)
+    parent_dir = os.path.dirname(in_file)
+    barcode = QCODE_REG.findall(basename)
+    if not barcode:
+        raise mtbutils.MTBdropboxerror('No QBiC Barcode found in {}'.format(in_file))
+    if len(set(barcode)) > 1:
+        raise mtbutils.MTBdropboxerror('More than one QBiC Barcode found in {}'.format(in_file))
+    tumor_dna_sample_id = barcode[0]
+    _, _, pbmc_sample_id = getentityandpbmc(tumor_dna_sample_id, transaction)
+
+    new_name = basename.replace(tumor_dna_sample_id, pbmc_sample_id)
+    new_path = os.path.join(parent_dir, new_name)
+    os.rename(in_file, new_path)
+
+    return new_path
+
+
+def proc_fastq(fastq_file, transaction):
+    """Register fastq as dataset in openBIS"""
+
+    # Check, if there are file pairs present (paired-end data!)
+    if len(fastq_file) != 2:
+        raise mtbutils.MTBdropboxerror('Expecting paired end reads files, found only {}'
+            .format(len(fastq_file)))
+    qbiccode_f1 = QCODE_REG.findall(os.path.basename(fastq_file[0]))
+    qbiccode_f2 = QCODE_REG.findall(os.path.basename(fastq_file[1]))
+    if not qbiccode_f1 or not qbiccode_f2:
+        raise mtbutils.MTBdropboxerror('No QBiC Barcode found in {}'.format(fastq_file))
+    if len(qbiccode_f1) > 1 or len(qbiccode_f2) > 1:
+        raise mtbutils.MTBdropboxerror('More than one QBiC Barcode found in {}'.format(fastq_file))
+    if qbiccode_f1[0] != qbiccode_f2[0]:
+        raise mtbutils.MTBdropboxerror('Found two different barcodes for read pair: {}<->{}'
+            .format(qbiccode_f1[0], qbiccode_f2[0]))
+
+    tumor_dna_sample_id = qbiccode_f1[0]
+    # Get space and project ids
+    space, project = space_and_project(tumor_dna_sample_id)
+
+    # Create new experiment id
+    experiments = transaction.getSearchService().listExperiments('/{}/{}'.format(space, project))
+    last_exp_id = get_last_exp_id(experiments)
+    new_exp_id = '/{space}/{project}/{project}E{number}'.format(
+        space=space, project=project, number=last_exp_id + COUNTER.newId())
+    new_sample_id = '/{space}/NGS{barcode}-{timestamp}'.format(
```

Comment:
timestamps in the sample code?

Reply:
my solution at the time to make it unique and allow for multiple runs